torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -669,13 +723,20 @@ def view_avoid_dupes_with_primals(t):
         traced_tangents = pytree.tree_map(
             view_avoid_dupes_with_primals, traced_tangents
         )
-        # See Note [Tangents must be contiguous]
-        traced_tangents = pytree.tree_map(
-            coerce_tangent,
-            traced_tangents,
-        )
-        user_outs = pytree.tree_map(from_fun, f_output_tangents)
 
+        output_tangents_start_idx = len(f_input_tangents)
+        output_tangents_end_idx = output_tangents_start_idx + len(f_output_tangents)
+        tangents_and_memory_formats = [
+            coerce_tangent_and_suggest_memory_format(
+                tt,
+                None
+                if (output_tangents_start_idx <= i and i < output_tangents_end_idx)
+                else torch.contiguous_format,
+            )
+            for i, tt in enumerate(traced_tangents)
```

Comment:
hmm, we used to always run `coerce_tangents` on all `traced_tangents`. Why are you filtering out some elements now? (if I'm reading this properly, I think it is trying to apply this logic to `f_input_tangents` and `f_output_tangents` but not `intermediate_bases`)

Reply:
Here the logic is that for 
`f_output_tangents` we deduce memory format with suggest_memory_format and coerce to that deduced memory_format.
`f_input_tangents`, `intermediate_bases` - the logic is as previous, we skip suggest memory format, forcing that format be contiguous